plausible/list_site_request.go

```diff
@@ -0,0 +1,44 @@
+package plausible
+
+import "strconv"
+
+type PaginationMeta struct {
+	After  string `json:"after"`
+	Before string `json:"before"`
+	Limit  int    `json:"limit"`
+}
+
+type ListSitesRequest struct {
+	After  string
+	Before string
+	Limit  int
+}
```

Comment:
Since these are objects about pagination, what do you think of giving them more generic names and move them to a new package called  `pagination`?

Also, related with the way pagination parameters are handled, I have a suggestion of using functional parameters.

We could have a type representing the pagination options like:

```go
type Paginator struct {
	After  string
	Before string
	Limit  int
}
```

Then we can define a type `Option` which represents a pagination option. An option here is a function that takes a paginator and modifies it.

```go
type Option func(*Paginator)
```

Here's how we can create a `Option` that sets the `After` field on the `Paginator`:

```go
func After(after string) PaginationOption {
    return func(p *Paginator) {
        p.After = after
    }
}
```

There would also be a `Before` and `Limit` built in the same way.

With this, we can make a function that takes an optional list of options and builds a paginator:

```go
func NewPaginator(options ...Option) *Paginator {
   paginator := &Paginator{}
    for _, opt := range options {
        opt(paginator)
    }
  return paginator
}
```

Then `ListSites` can contain just take an optional list of pagination options:

```go
func ListSites(pagination ...pagination.Option) {
   // ...
}
```

The inside `ListSites` we build a paginator from the options by calling `NewPaginator`. To convert a paginator to query arguments, probably it's best to add the `toQueryArgs` method to the `Paginator` itself.

One great advantage of doing this, is that we can then call `ListSites` without any arguments:

```go
sites, err := client.ListSites()
```

Or if we just want to set a single option:

```go
sites, err := client.ListSites(pagination.Limit(20))
```

Or several:

```go
sites, err := client.ListSites(
  pagination.After("awebsite"),
  pagination.Before("otherwebsite"),
  pagination.Limit(20),
)
```

Reply:
It would be cool to also add documentation comments for all these new types.